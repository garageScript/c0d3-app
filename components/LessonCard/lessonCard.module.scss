@use '../../scss/variables.module';

.lesson-card__container_inprogress {
  border-width: 2px;
}
.lesson-card__title {
  font-size: 1.5rem;
  line-height: 1.3;
  margin-bottom: 0.75rem;
}

.lesson-card__badge {
  align-self: flex-start;
  padding-left: 1em;
  padding-right: 1em;
  margin-left: auto;
  background-color: variables.$success;
  @media (max-width: 768px) {
    background-color: transparent;
    color: #00cf92;
    margin-left: auto;
    & > svg {
      width: 30px;
      height: 30px;
      stroke-width: 3px;
    }
  }
}

.lesson-card__icon-text {
  font-size: 0.8rem;
  line-height: 1.4;
}
.lesson-card__button {
  font-size: 13px;
  min-width: 134px;
}
.lesson-card__image {
  object-fit: contain;
}
.lesson-card__description {
  flex: 1;
}
