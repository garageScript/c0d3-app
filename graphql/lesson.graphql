query Lesson {
  lessons {
    id
    title
  }
}