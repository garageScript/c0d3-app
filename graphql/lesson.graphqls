type Lesson {
  id: String
  title: String
}

type Query {
  lessons: Lesson!
}