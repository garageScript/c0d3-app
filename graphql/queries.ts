import { gql } from 'apollo-boost'

export const GET_LESSONS = gql`
  query Lessons {
    curriculumStatus {
      id
      description
      title
      order
      docUrl
      challenges {
        id
      }
      currentUser {
        userLesson {
          isEnrolled
          isTeaching
          isPassed
        }
      }
    }
    lessons {
      id
      title
      description
      order
      docUrl
      challenges {
        id
        description
        title
        order
      }
    }
  }
`

export const GET_LESSON = gql`
  query lessonInfo(
    $lessonInfo: LessonId
    $lessonUserInfo: LessonUserId
    $lessonStatus: LessonId
  ) {
    lessonInfo(input: $lessonInfo) {
      id
      title
      description
      order
      docUrl
      chatUrl
      challenges {
        id
        description
        title
        order
      }
    }
    userSubmissions(input: $lessonUserInfo) {
      id
      status
      mrUrl
      diff
      viewCount
      comment
      challengeId
      reviewer {
        id
        username
      }
      createdAt
      updatedAt
    }
    lessonStatus(input: $lessonStatus) {
      isPassed
      isTeaching
      isEnrolled
    }
  }
`

export const GET_SUBMISSIONS = gql`
  query submissions($in: LessonId) {
    submissions(input: $in) {
      status
    }
  }
`
