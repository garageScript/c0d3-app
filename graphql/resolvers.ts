import { login, logout, signup, isTokenValid } from './resolvers/authController'
import {
  createChallenge,
  updateChallenge,
  challenges
} from './resolvers/challengesController'
import { userInfo } from './resolvers/userInfoController'
import { addAlert, removeAlert } from './resolvers/alertController'
import { reqPwReset, changePw } from './resolvers/passwordController'
import { addComment } from './resolvers/addComment'
import {
  createSubmission,
  acceptSubmission,
  rejectSubmission,
  submissions
} from './resolvers/submissionController'
import { setStar } from './resolvers/starsController'
import { alerts } from './resolvers/alerts'
import { lessons } from './resolvers/lessons'
import { session } from './resolvers/session'
import { allUsers } from './resolvers/allUsers'
import { getLessonMentors } from './resolvers/getLessonMentors'
import { changeAdminRights } from './resolvers/adminController'
import { createLesson, updateLesson } from './resolvers/lessonsController'
import { getPreviousSubmissions } from './resolvers/getPreviousSubmissions'
import { deleteComment } from './resolvers/deleteComment'
import { unlinkDiscord } from './resolvers/unlinkDiscord'
import { editComment } from './resolvers/editComment'
import {
  addModule,
  modules,
  updateModule,
  deleteModule
} from './resolvers/moduleCrud'
import {
  exercises,
  addExercise,
  updateExercise,
  deleteExercise,
  flagExercise,
  removeExerciseFlag,
  removeExercise
} from './resolvers/exerciseCrud'
import {
  exerciseSubmissions,
  addExerciseSubmission
} from './resolvers/exerciseSubmissionCrud'
import {
  getExerciseComments,
  addExerciseComment,
  getChildComments,
  editExerciseComment
} from './resolvers/exerciseCommentCrud'
import { updateUserNames, updateUserPassword } from './resolvers/userDataCrud'

export default {
  Query: {
    submissions,
    getLessonMentors,
    allUsers,
    isTokenValid,
    userInfo,
    lessons,
    exercises,
    exerciseSubmissions,
    modules,
    session,
    alerts,
    getPreviousSubmissions,
    challenges,
    getExerciseComments,
    getChildComments
  },

  Mutation: {
    setStar,
    changePw,
    changeAdminRights,
    createSubmission,
    acceptSubmission,
    rejectSubmission,
    createLesson,
    updateLesson,
    addExercise,
    updateExercise,
    deleteExercise,
    addExerciseSubmission,
    login,
    logout,
    signup,
    addAlert,
    removeAlert,
    reqPwReset,
    createChallenge,
    updateChallenge,
    addModule,
    updateModule,
    deleteModule,
    addComment,
    editComment,
    deleteComment,
    flagExercise,
    removeExerciseFlag,
    removeExercise,
    unlinkDiscord,
    addExerciseComment,
    editExerciseComment,
    updateUserNames,
    updateUserPassword
  }
}
