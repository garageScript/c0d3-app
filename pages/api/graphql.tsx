import { ApolloServer, gql } from 'apollo-server-micro'

const typeDefs = gql`
  type Query {
    username: String
    hello: String
  }
`

const resolvers = {
  Query: {
    username(_parent: any, _args: any, ctx: any) {
      return ctx.user
    },
    hello() {
      return 'hello'
    }
  }
}

const context = ({ req }: { req: any }) => req.cookies

const apolloServer = new ApolloServer({ typeDefs, resolvers, context })

export const config = {
  api: {
    bodyParser: false
  }
}

export default apolloServer.createHandler({ path: '/api/graphql' })
