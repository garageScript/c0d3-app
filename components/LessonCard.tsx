import React from 'react'
import { useQuery } from '@apollo/react-hooks'
import { CheckCircle } from 'react-feather'
import { GET_SUBMISSIONS } from '../graphql/queries'
import {
  Props,
  ReviewButtonProps,
  ReviewCountProps
} from '../@types/lessonCard'
import NavLink from './NavLink'
import '../scss/lessonCard.scss'

const ReviewCount: React.FC<ReviewCountProps> = props => {
  if (props.shouldNotGetCount) {
    return null
  }

  const { loading, data } = useQuery(GET_SUBMISSIONS, {
    variables: {
      in: {
        id: `${props.lessonId}`
      }
    }
  })

  if (loading) {
    return (
      <div className="spinner-border spinner-border-sm mx-1" role="status" />
    )
  }

  if (!data) {
    return null
  }
  const pendingSubmissionsCount = data.submissions.reduce(
    (acc: number, val: any) => {
      if (val.status === 'open') {
        acc = acc + 1
        return acc
      }
      return acc
    },
    0
  )

  return <span> {pendingSubmissionsCount} </span>
}

const ReviewButton: React.FC<ReviewButtonProps> = props => {
  if (!props.isCompleted) {
    return null
  }
  return (
    <a
      href={props.reviewUrl}
      className="btn btn-sm bg-primary text-white float-right mb-2 mr-2"
    >
      Review
      <ReviewCount
        shouldNotGetCount={props.shouldNotGetCount}
        lessonId={props.lessonId}
      />
      Submissions
    </a>
  )
}

const LessonCard: React.FC<Props> = props => {
  const containerClass =
    props.currentState === 'inProgress'
      ? 'lesson-card__container_inprogress border-primary'
      : 'border-0'

  return (
    <div className={`card shadow-sm mt-3 ${containerClass}`}>
      <div className="d-flex p-2">
        <img
          src={`/curriculumAssets/lessonCoversSvg/${props.coverImg}`}
          alt={props.coverImg}
        />

        <div className="w-100 pl-4">
          {props.currentState === 'completed' && (
            <span className="badge badge-pill badge-success float-right mt-2 mr-2 p-2 d-flex align-items-center">
              <CheckCircle style={{ height: '15px' }} />
              <span className="mx-1">COMPLETED</span>
            </span>
          )}
          <div>
            <h4 className="lesson-card__title font-weight-bold mt-3">
              <NavLink
                as={`/curriculum/${props.lessonId}`}
                path="/curriculum/[lesson]"
                text={props.title}
              />
            </h4>
            <div>
              <div className="d-inline-block mr-4">
                <img
                  className="mr-2"
                  src="/curriculumAssets/icons/icon-challenge.svg"
                  alt="icon-challenge"
                />
                <span className="lesson-card__icon-text">
                  {props.challengeCount} CHALLENGES
                </span>
              </div>
            </div>
            <p className="lesson-card__description mt-2">{props.description}</p>
          </div>
          <ReviewButton
            isCompleted={props.currentState === 'completed'}
            reviewUrl={props.reviewUrl}
            shouldNotGetCount={props.shouldNotGetCount}
            lessonId={props.lessonId}
          />
        </div>
      </div>

      {props.currentState === 'inProgress' && (
        <div className="p-2 bg-primary">
          <NavLink
            path={props.docUrl}
            text="Start Lesson"
            className="lesson-card__button btn btn-light mr-2 my-1 text-primary"
            external
          />
          <NavLink
            path={props.challengesUrl}
            text="View Challenges"
            className="lesson-card__button btn bg-primary my-1 text-white border border-white"
          />
        </div>
      )}
    </div>
  )
}

export default LessonCard
