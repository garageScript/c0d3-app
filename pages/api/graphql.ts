import { withSentry } from '@sentry/nextjs'
import { ApolloServer } from 'apollo-server-micro'
import { createRouter, expressWrapper } from 'next-connect'
import resolvers from '../../graphql/resolvers'
import typeDefs from '../../graphql/typeDefs'
import { apolloLogPlugin } from '../../helpers/apolloLogPlugin'
import loggingMiddleware from '../../helpers/middleware/logger'
import userMiddleware from '../../helpers/middleware/user'
import sessionMiddleware from '../../helpers/middleware/session'

// VERCEL_ENV is a reserved env key by Vercel that specify the deployment type: "preview", "production", or "deployment"
const isDevEnv =
  process.env.VERCEL_ENV === 'preview' || process.env.NODE_ENV === 'development'

const router = createRouter()
const apolloServer = new ApolloServer({
  typeDefs,
  resolvers,
  context: ({ req, res }) => ({ req, res }), // This lets GraphQL have access to sessions
  /* Syncs server schema (used for server static generation) and api route server settings.
  By default apolloServer accepts uploads, while schema-generated server does not.*/
  uploads: false,
  plugins: [apolloLogPlugin],
  ...(isDevEnv && {
    playground: {
      settings: {
        'request.credentials': 'include'
      }
    },
    introspection: isDevEnv
  })
})

const graphQLHandler = apolloServer.createHandler({ path: '/api/graphql' })

router
  .use(expressWrapper(loggingMiddleware))
  .use(expressWrapper(sessionMiddleware()))
  .use(expressWrapper(userMiddleware))
  .get('/api/graphql', graphQLHandler)
  .post('/api/graphql', graphQLHandler)

// Config needs to be exported because we are changing default values in Next.JS API https://nextjs.org/docs/api-routes/api-middlewares
// This is required for GraphQL to work properly  https://blog.logrocket.com/building-a-graphql-server-in-next-js/
export const config = {
  api: {
    bodyParser: false
  }
}

export default withSentry(router.handler())
