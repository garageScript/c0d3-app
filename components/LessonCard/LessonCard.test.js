jest.mock('@apollo/client')
import { useQuery } from '@apollo/client'
import * as React from 'react'
import LessonCard from './LessonCard'
import { render, screen } from '@testing-library/react'
import { SubmissionStatus } from '../../graphql'

// Imported to be able to use expect(...).toBeInTheDocument()
import '@testing-library/jest-dom'

describe('Lesson Card Complete State', () => {
  const props = {
    challengesUrl: '/challengeUrl',
    reviewUrl: '/reviewUrl'
  }
  test('Should render lessonCard with null if no data', async () => {
    useQuery.mockReturnValue({
      data: null
    })

    const { container } = render(
      <LessonCard {...props} currentState="completed" />
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render lessonCard with loading...', async () => {
    useQuery.mockReturnValue({
      loading: true
    })

    const { container } = render(
      <LessonCard {...props} currentState="completed" />
    )
    expect(container).toMatchSnapshot()
  })

  test('Should render lessonCard with submission count', async () => {
    useQuery.mockReturnValue({
      data: {
        submissions: [
          {
            status: SubmissionStatus.Open
          },
          {
            status: SubmissionStatus.Open
          },
          {
            status: SubmissionStatus.NeedMoreWork
          }
        ]
      }
    })

    render(<LessonCard {...props} currentState="completed" />)

    expect(
      screen.getAllByTestId('review-submissions-count')[0]
    ).toBeInTheDocument()
  })

  test('Should render lessonCard with no submission count', async () => {
    useQuery.mockReturnValue({
      data: {
        submissions: []
      }
    })

    render(<LessonCard {...props} currentState="completed" />)

    expect(
      screen.queryByTestId('review-submissions-count')
    ).not.toBeInTheDocument()
  })
})

describe('Lesson Card', () => {
  const props = {
    challengesUrl: '/challengeUrl',
    reviewUrl: '/reviewUrl'
  }
  test('Should render loading card when loading', async () => {
    useQuery.mockReturnValue({
      loading: true
    })

    const { container } = render(<LessonCard {...props} />)
    expect(container).toMatchSnapshot()
  })
  test('Should render ellpsis when no data is present', async () => {
    useQuery.mockReturnValue({
      loading: false
    })

    const { container } = render(<LessonCard {...props} />)
    expect(container).toMatchSnapshot()
  })
  test('Should render lessonCard with no submission count', async () => {
    useQuery.mockReturnValue({
      data: {
        submissions: []
      }
    })

    const { container } = render(<LessonCard {...props} />)
    expect(container).toMatchSnapshot()
  })
  test('Should render lessonCard with inProgress currentState prop', async () => {
    const { container } = render(
      <LessonCard {...props} currentState="inProgress" />
    )
    expect(container).toMatchSnapshot()
  })
  test('Should render lessonCard with lessonId prop', async () => {
    const { container } = render(<LessonCard {...props} lessonId={4} />)
    expect(container).toMatchSnapshot()
  })
})
