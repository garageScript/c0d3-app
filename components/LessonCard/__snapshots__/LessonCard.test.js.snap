// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lesson Card Complete State Should render lessonCard with loading... 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 border-0"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
          <span
            class="lesson-card__badge badge rounded-pill p-0"
          >
            <svg
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              class="mx-1 d-none d-md-block"
            >
              COMPLETED
            </span>
          </span>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
      <a
        class="btn btn-sm btn-primary text-white float-end mb-2 me-2 me-0"
        data-testid="review-submissions-count"
        href="/reviewUrl"
      >
        Review 
        <div
          class="spinner-border spinner-border-sm mx-1"
          role="status"
        />
         Submissions
      </a>
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
          <span
            class="lesson-card__badge badge rounded-pill mt-2 me-2 py-2 d-flex align-items-center"
          >
            <svg
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              class="mx-1 d-none d-md-block"
            >
              COMPLETED
            </span>
          </span>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
        <a
          class="btn btn-sm btn-primary text-white float-end mb-2 me-2 "
          data-testid="review-submissions-count"
          href="/reviewUrl"
        >
          Review 
          <div
            class="spinner-border spinner-border-sm mx-1"
            role="status"
          />
           Submissions
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lesson Card Complete State Should render lessonCard with null if no data 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 border-0"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
          <span
            class="lesson-card__badge badge rounded-pill p-0"
          >
            <svg
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              class="mx-1 d-none d-md-block"
            >
              COMPLETED
            </span>
          </span>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
          <span
            class="lesson-card__badge badge rounded-pill mt-2 me-2 py-2 d-flex align-items-center"
          >
            <svg
              fill="none"
              height="15"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M22 11.08V12a10 10 0 1 1-5.93-9.14"
              />
              <polyline
                points="22 4 12 14.01 9 11.01"
              />
            </svg>
            <span
              class="mx-1 d-none d-md-block"
            >
              COMPLETED
            </span>
          </span>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lesson Card Should render ellpsis when no data is present 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 border-0"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lesson Card Should render lessonCard with inProgress currentState prop 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 lesson-card__container_inprogress border-primary"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
      </div>
    </div>
    <div
      class="p-2 bg-primary"
    >
      <a
        class="lesson-card__button btn btn-primary my-1 text-white border border-white noUnderline"
        href="/challengeUrl"
      >
        View Challenges
      </a>
    </div>
  </div>
</div>
`;

exports[`Lesson Card Should render lessonCard with lessonId prop 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 border-0"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lesson Card Should render lessonCard with no submission count 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 border-0"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Lesson Card Should render loading card when loading 1`] = `
<div>
  <div
    class="card shadow-sm mt-3 border-0"
  >
    <div
      class="d-flex d-sm-none flex-column px-3"
    >
      <h4
        class="lesson-card__title fw-bold mt-3 "
      >
        <div
          class="d-flex justify-content-center"
        >
          <div
            class="align-self-center"
          >
            <a
              class=" noUnderline"
              href="/challengeUrl"
            />
          </div>
        </div>
      </h4>
      <img
        class="rounded"
        height="165"
        layout="intrinsic"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div>
        <div
          class="d-inline-block mt-3"
        >
          <img
            alt="icon-challenge"
            class="me-2"
            src="/assets/curriculum/icons/icon-challenge.svg"
          />
          <span
            class="lesson-card__icon-text"
          >
             CHALLENGES
          </span>
        </div>
      </div>
      <p
        class="mt-2"
      />
    </div>
    <div
      class="d-none d-sm-flex p-2"
    >
      <img
        class="align-self-center rounded"
        height="165"
        src="/assets/curriculum/undefined"
        width="116"
      />
      <div
        class="lesson-card__description ps-4 d-none d-sm-block"
      >
        <div
          class="d-flex"
        >
          <h4
            class="lesson-card__title fw-bold mt-3"
          >
            <a
              class="text-primary link"
              href="/challengeUrl"
            />
          </h4>
        </div>
        <div>
          <div>
            <div
              class="d-inline-block me-4"
            >
              <img
                alt="icon-challenge"
                class="me-2"
                src="/assets/curriculum/icons/icon-challenge.svg"
              />
              <span
                class="lesson-card__icon-text"
              >
                 CHALLENGES
              </span>
            </div>
          </div>
          <p
            class="mt-2"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
